### DISCUSSION TITLE
[QUESTION] Is possible to programatically install autocompletion?

### DISCUSSION THREAD

**Discussion by ferreteleco** [fastapi/typer#1578] [Questions]
### First check

* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the Typer documentation, with the integrated search.
* [X] I already searched in Google "How to X in Typer" and didn't find any information.
* [X] I already searched in Google "How to X in Click" and didn't find any information.

### Description

Is it possible / would it be possible to programatically install completions (for example during pip install <package>) in order to exclude --install-completion / --show-completion from CLI options?

### Additional context

The use case is the following: an app delivered to a customer as much auto-configured as possible, since its technical knowledge / ability is limited and the motto for me here is providing software as simple as possible, therefore keeping only bare minimum configuration options without losing functionality.

Scenario:

cmd / bash shells (Win10 and Ubuntu 20.04).

Thanks in advance!

**Comment by pmav99** [comment]
if you want to avoid having the `--install-completion` option appearing in `--help` appearing, you should create the `Typer` object with `add_completion=False`:

``` python
import typer

app = typer.Typer(add_completion=False)
```

That being said, I don't know if there is a clean way to run a "post-install" hook when using `pip install`. Nevertheless, the job of actually installing the completions seems kind of out of scope for something like pip (there are too many variables, OS, shell flavors, shell versions, user configuration, permissions etc). If you have need of this maybe you can just create a bash script containing:

``` bash
#!/usr/bin/env bash

set -euo pipefail

pip install my_app
my_app --show-completion > /where/this/should/be/placed
```

**Comment by ssbarnea** [comment]
I just want to emphasise that I agree with the proposed "just-works" behaviour would a very good selling-point of Typer. If you endup having to write documentation on how to enable completion for various shells, you already failed a little bit on UX side.

I think that an ideal app should prompt for optional installation of auto-completion hooks the first time is called. That approach should be seen as a decent trade-off.

**Comment by liammac** [comment]
I think I'm in the same boat here. I actually just want a way to initiate  `<script> --install-completion` from within my code somewhere without shelling out. I have a `<script> config` argument where I do some other setup and config I would really love to be able to add the `--install-completion` step there. Can we just get a function exposed to run that?  I can't quite unpack how it gets done now looking at the source
